impeller/entity/entity.cc

```diff
@@ -110,7 +110,7 @@ uint32_t Entity::GetNewClipDepth() const {
 static const Scalar kDepthEpsilon = 1.0f / std::pow(2, 18);
 
 float Entity::GetShaderClipDepth() const {
-  return new_clip_depth_ * kDepthEpsilon;
+  return std::clamp(new_clip_depth_ * kDepthEpsilon, 0.0f, 1.0f);
```

Comment:
Is this just defensive coding or did you actually run into this?

Reply:
Oh, I see the max-uint setting below. Nevermind.

Reply:
Yeah not sure about keeping the max int pattern... thinking about replacing with a `SetMaxDepth` or addding a `kMaxDepth` sentinel to make the code easier to read.